samza-core/src/main/scala/org/apache/samza/coordinator/JobModelManager.scala

```diff
@@ -288,15 +290,16 @@ object JobModelManager extends Logging {
     * @param streamMetadataCache required to query the partition metadata of the input streams.
     * @return the input SystemStreamPartitions of the job.
     */
-  private def getMatchedInputStreamPartitions(config: Config, streamMetadataCache: StreamMetadataCache): Set[SystemStreamPartition] = {
+  private def getMatchedInputStreamPartitions(config: Config, streamMetadataCache: StreamMetadataCache,
+    classLoader: ClassLoader): Set[SystemStreamPartition] = {
     val allSystemStreamPartitions = getInputStreamPartitions(config, streamMetadataCache)
     config.getSSPMatcherClass match {
       case Some(s) =>
         val jfr = config.getSSPMatcherConfigJobFactoryRegex.r
         config.getStreamJobFactoryClass match {
           case Some(jfr(_*)) =>
             info("before match: allSystemStreamPartitions.size = %s" format allSystemStreamPartitions.size)
-            val sspMatcher = Util.getObj(s, classOf[SystemStreamPartitionMatcher])
+            val sspMatcher = ReflectionUtil.getObj(classLoader, s, classOf[SystemStreamPartitionMatcher])
```

Comment:
not related to your change, but since there are some reorgs/clean up, can we also change the variable `s` to something meaningful?